### ISSUE TITLE
Can Not Start Calico Node on v0.6.0

### ISSUE THREAD

**Issue by ninjadq** [projectcalico/calico-docker#486]
When I using this command to start a node
```bash
sudo calicoctl node --libnetwork
```
It returns that
```
No IP provided. Using detected IP: 172.17.8.101
Unexpected error executing command.

Traceback (most recent call last):
  File "<string>", line 102, in <module>
  File "/code/build/calicoctl/out00-PYZ.pyz/calico_ctl.node", line 175, in node
  File "/code/build/calicoctl/out00-PYZ.pyz/calico_ctl.node", line 219, in node_start
  File "/code/build/calicoctl/out00-PYZ.pyz/calico_ctl.node", line 408, in warn_if_hostname_conflict
TypeError: object of type 'NoneType' has no len()
```
It seems that in the function [warn_if_hostname_conflict](https://github.com/projectcalico/calico-docker/blob/master/docs/getting-started/libnetwork/Demonstration.md) `docker_client.containers(filters={'name': 'calico-node'})` returned a None.

I'm Using Centos 7 and docker-1.9experiments, and the calicoctl version is v6.0.0. Can someone help me to resolve this issue.  :bow:

**Comment by TrimBiggs** [comment]
Hi @ninjadq 

Thanks for bringing up this issue.  I'm looking into it now, and I've just reproduced the problem locally.  I'll keep you updated on a fix for this when I've made more progress.

**Comment by TrimBiggs** [comment]
I reproduced this earlier using the latest Docker experimental.  When I use an older version of Docker experimental, I don't see this issue, so it seems that some Docker commit has changed the return value for this call.

I've made a fix for this in PR #487.  For now, you can clone the repository once the change is in and run `sudo make binary` to create the `calicoctl` binary at `/path/to/calico-docker/dist/calicoctl`.

Keep an eye out for a potential release this week as well, which will contain the fix.

**Comment by TrimBiggs** [comment]
Fixed by #487
